modules/rosetta/src/main/scala/org/tessellation/rosetta/domain/Amount.scala

```diff
@@ -1,31 +1,60 @@
 package org.tessellation.rosetta.domain
 
 import org.tessellation.ext.derevo.magnoliaCustomizable.snakeCaseConfiguration
-import org.tessellation.rosetta.domain.currency.Currency
+import org.tessellation.rosetta.domain.currency.{Currency, DAG}
+import org.tessellation.schema.transaction.{TransactionAmount, TransactionFee}
 
-import derevo.cats.eqv
+import derevo.cats.{eqv, show}
 import derevo.circe.magnolia._
 import derevo.derive
 import eu.timepit.refined.api.Refined
 import eu.timepit.refined.auto._
 import eu.timepit.refined.boolean.Not
 import eu.timepit.refined.cats._
 import eu.timepit.refined.generic.Equal
+import eu.timepit.refined.numeric.{NonNegative, Positive}
+import eu.timepit.refined.refineV
 import io.circe.refined._
 import io.estatico.newtype.macros.newtype
 
+// TODO should we rename Amount to NonZeroAmount ?
+//  it would be clearer and not clash with schema.balance.Amount
+
+// TODO should we unpack Amount.value and make it Refined[Long, NonZero] where type NonZero = Not[Equal[0L]] ?
+//  I'm not clear what we gain by the extra wrapping other than more work by callers
```

Comment:
Instead of writing TODOs in the code, we:
- comment the part of code on the review (with mention to protocol team) so we can discuss it further
- create a ticket in jira if there is actually something left for TODO